sql/expression/comparison.go

```diff
@@ -178,12 +179,25 @@ func (e *Equals) String() string {
 // Regexp is a comparison that checks an expression matches a regexp.
 type Regexp struct {
 	comparison
-	r regex.Matcher
+	pool   *sync.Pool
+	cached bool
 }
 
 // NewRegexp creates a new Regexp expression.
 func NewRegexp(left sql.Expression, right sql.Expression) *Regexp {
-	return &Regexp{newComparison(left, right), nil}
+	var cached bool
+	Inspect(right, func(e sql.Expression) bool {
```

Comment:
I think this functionality is already on Like, can you check to see if this might be reused?